src/classes/QueryBuilder.cls

```diff
@@ -166,11 +180,12 @@ public without sharing class QueryBuilder extends NebulaCore implements IQueryBu
 
         this.addLogEntry(query);
 
-        return query;
+        return this.query;
     }
 
     private void addCommonQueryFields() {
         if(!NebulaSettings.repositorySettings.IncludeCommonFields__c) return;
+        if(!this.shouldAddCommonFields) return;
```

Comment:
I moved the check for 'adding common fields' to a custom setting - do you think it's useful to have both a custom setting and a property/method to control adding the common fields?